packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -26,16 +28,42 @@ function isGroupContents(opContents: any): opContents is IGroupedBatchMessageCon
 	return opContents?.type === OpGroupingManager.groupedBatchOp;
 }
 
+export interface OpGroupingManagerConfig {
+	readonly groupedBatchingEnabled: boolean;
+	readonly opCountThreshold: number;
+	readonly reentrantBatchGroupingEnabled: boolean;
+}
+
 export class OpGroupingManager {
 	static readonly groupedBatchOp = "groupedBatch";
+	private readonly logger;
 
-	constructor(private readonly groupedBatchingEnabled: boolean) {}
+	constructor(
+		private readonly config: OpGroupingManagerConfig,
+		logger: ITelemetryBaseLogger,
+	) {
+		this.logger = createChildLogger({ logger, namespace: "OpGroupingManager" });
+	}
 
 	public groupBatch(batch: IBatch): IBatch {
-		if (batch.content.length < 2 || !this.groupedBatchingEnabled) {
+		if (!this.shouldGroup(batch)) {
 			return batch;
 		}
 
+		assert(
+			batch.content.length >= this.config.opCountThreshold,
+			"Expecting to always group batches with op counts larger than the threshold",
+		);
+		if (this.config.opCountThreshold >= 1000) {
```

Comment:
This should be safe, we currently have an event (`fluid:telemetry:Batching:LengthTooBig`) which fires for batches with over 1000 ops. It is not very chatty.